gc.go

```diff
@@ -33,7 +33,87 @@ type Config struct {
 	WatchDir              string `json:"watch_dir"`
 }
 
+func setupConfig() *Config {
+	var config Config
+
+	configFiles := []string{"${XDG_CONFIG_HOME}/gc/config",
+		"${HOME)/.config/gcrc",
+		"${HOME}/.gcrc"}
+
+	var path string
+	for _, path = range configFiles {
+		path = os.ExpandEnv(path)
+		_, err := os.Stat(path)
+		if err == nil {
+			configFile, err := os.Open(path)
+
+			if err != nil {
+				panic(err)
+			}
+
+			err = json.NewDecoder(configFile).Decode(&config)
+
+			if err != nil {
+				panic(err)
+			}
+
+			break
+		} else {
+			path = ""
+		}
+	}
+
+	if argNoInteractive && path == "" {
+		panic("No config file found and started in non-interactive mode\n")
+	}
+
+	if !argNoInteractive {
+		if argUsername != "" {
+			config.GarminConnectUsername = argUsername
+		} else if config.GarminConnectUsername == "" {
+			fmt.Print("Garmin Connect Username: ")
+			reader := bufio.NewReader(os.Stdin)
+			username, err := reader.ReadString('\n')
+			if err != nil {
+				panic(err)
+			}
+			config.GarminConnectUsername = strings.TrimSpace(username)
+		}
+
+		if config.GarminConnectPassword == "" {
+			fmt.Print("Garmin Connect Password: ")
+			bytePassword, err := terminal.ReadPassword(int(syscall.Stdin))
+			fmt.Print("\n")
+			if err != nil {
+				panic(err)
+			}
+			config.GarminConnectPassword = string(bytePassword)
+		}
+
+		if argWatchDir != "" {
+			config.WatchDir = argWatchDir
+		} else if config.WatchDir == "" {
+			fmt.Print("Watch mount directory: ")
+			reader := bufio.NewReader(os.Stdin)
+			watchdir, err := reader.ReadString('\n')
+			if err != nil {
+				panic(err)
+			}
+			config.WatchDir = strings.TrimSpace(watchdir)
+		}
+	}
+
+	if config.GarminConnectUsername == "" ||
+		config.GarminConnectPassword == "" ||
+		config.WatchDir == "" {
+		panic("Option(s) missing. Aborting")
```

Comment:
Do we want to `panic` or exit cleanly with a nice error message with `os.Exit(1)`?